### PR TITLE
[9.1.x] ISPN-8557 RESTful DELETE and PUT should always work

### PR DESCRIPTION
**Pull request by tristantarrant**
- reordered the test conditions so that metadata is only considered when etags are present

https://issues.jboss.org/browse/ISPN-8557

### CODE REVIEW

**Comment by gustavonalle** [comment]
@tristantarrant Puzzled why we didn't have a test for that (simple) case, can you add it?

**Comment by tristantarrant** [comment]
@gustavonalle added a test

File: server/rest/src/main/java/org/infinispan/rest/operations/CacheOperations.java [hunk 1]
```diff
@@ -206,21 +206,23 @@ public InfinispanResponse deleteCacheValue(InfinispanRequest request) throws Res
...
-                  response.status(HttpResponseStatus.OK);
-                  if (useAsync.isPresent() && useAsync.get()) {
-                     restCacheManager.getCache(cacheName).removeAsync(key);
+            Optional<String> clientEtag = request.getEtagIfNoneMatch();
+            if (clientEtag.isPresent()) {
+               Metadata meta = entry.getMetadata();
+               if (meta instanceof MimeMetadata) {
+                  String etag = calcETAG(ice.getValue(), ((MimeMetadata) meta).contentType());
+
+                  if (clientEtag.map(t -> t.equals(etag)).orElse(true)) {
```
Comment (danberindei) [inline server/rest/src/main/java/org/infinispan/rest/operations/CacheOperations.java]: We're in the `clientEtag.isPresent()` branch, so this is equivalent to `clientEtag.get().equals(etag)`

Comment (tristantarrant) [reply server/rest/src/main/java/org/infinispan/rest/operations/CacheOperations.java]: Fixed

**Comment by tristantarrant** [comment]
For the TO State transfer test failures I've backported https://github.com/infinispan/infinispan/pull/5622

**Comment by tristantarrant** [comment]
CI has 3 unrelated failures.

**Pull request merged**